drivers/exec/driver.go

```diff
@@ -688,3 +735,12 @@ func (d *Driver) ExecTaskStreamingRaw(ctx context.Context,
 
 	return handle.exec.ExecStreaming(ctx, command, tty, stream)
 }
+
+func getUsername(cfg *drivers.TaskConfig) string {
```

Comment:
When I see the failing E2E test and look at this function, it makes me think that instead of having to call this function every time we want `cfg.User`, that we should _set_ `cfg.User` like this:

```go
if cfg.User == "" {
	cfg.User = "nobody"
}
```

somewhere closer to the top of `StartTask` (maybe even in `validateUserIds`?). That way you only have to do this once and now all downstream consumers of `cfg.User`, like `(*TaskConfig).Validate` know that it's definitely non-empty.

Reply:
+1 to this idea - will implement